astropy/table/table.py

```diff
@@ -3968,6 +3968,10 @@ def group_by(self, keys):
         -------
         out : `~astropy.table.Table`
             New table with groups set
+
+        Notes
+        -----
+        The underlying sorting algorithm is guaranteed stable.
```

Comment:
Not all users will know what a "stable" sort is.
```suggestion
        The underlying sorting algorithm is guaranteed stable, meaning that the
        original table order is preserved within each group.
```